### ISSUE TITLE
A11Y Proxy Container doesn't update when chart data updates

### ISSUE THREAD

**Issue by ashleyryan** [highcharts/highcharts#15902]
#### Expected behaviour

When the series of a chart changes, the hidden A11Y buttons/elements should update as well.

#### Actual behaviour
The A11Y buttons still displays the original series.

#### Live demo with steps to reproduce
http://jsfiddle.net/BlackLabel/kz5yneb6/

Open the console, click the button and notice that the original series is still listed

#### Product version
Highcharts 9.1.1 with Accessibility Module

#### Affected browser(s)
Chrome

Note: I originally reported this in [highcharts-react](https://github.com/highcharts/highcharts-react/issues/303) and Paweł sent me here.
